site/src/components/components/LivePreview.tsx

```diff
@@ -92,6 +93,98 @@ export const LivePreview: FC<LivePreviewProps> = ({
   // somewhere), then fallback to using own state
   const showCode = contextOnShowCodeToggle ? contextShowCode : ownShowCode;
 
+  const editInStackBlitz = () => {
+    const PACKAGE_JSON = {
+      name: "react-ts",
+      version: "0.0.0",
+      private: true,
+      dependencies: {
+        "@fontsource/open-sans": "^4.5.13",
+        "@fontsource/pt-mono": "^4.5.11",
+        "@salt-ds/core": "latest",
+        "@salt-ds/countries": "latest",
+        "@salt-ds/icons": "latest",
+        "@salt-ds/lab": "latest",
+        "@salt-ds/theme": "latest",
+        "@types/react": "^18.3.0",
+        "@types/react-dom": "^18.3.0",
+        react: "^18.3.0",
+        "react-dom": "^18.3.0",
+      },
+      scripts: {
+        start: "react-scripts start",
+        build: "react-scripts build",
+        test: "react-scripts test --env=jsdom",
+        eject: "react-scripts eject",
+      },
+      devDependencies: {
+        "react-scripts": "latest",
+      },
+    };
+    StackBlitzSDK.openProject(
+      // Payload
+      {
+        // REQUIRED: specify dependencies
+        dependencies: PACKAGE_JSON.dependencies,
+        files: {
+          "index.html": `<div id="root"></div>`,
+          "index.tsx": `import * as React from 'react';
+          import { StrictMode } from 'react';
+          import { createRoot } from 'react-dom/client';
+          
+          import '@salt-ds/theme/index.css';
+          import '@fontsource/open-sans/300.css';
+          import '@fontsource/open-sans/300-italic.css';
+          import '@fontsource/open-sans/400.css';
+          import '@fontsource/open-sans/400-italic.css';
+          import '@fontsource/open-sans/500.css';
+          import '@fontsource/open-sans/500-italic.css';
+          import '@fontsource/open-sans/600.css';
+          import '@fontsource/open-sans/600-italic.css';
+          import '@fontsource/open-sans/700.css';
+          import '@fontsource/open-sans/700-italic.css';
+          import '@fontsource/open-sans/800.css';
+          import '@fontsource/open-sans/800-italic.css';
+          import '@fontsource/pt-mono';
+          
+          import App from './App';
+          import { SaltProvider } from '@salt-ds/core';
+          
+          const rootElement = document.getElementById('root');
+          const root = createRoot(rootElement);
+          
+          root.render(
+            <StrictMode>
+              <SaltProvider>
+                <App />
+              </SaltProvider>
+            </StrictMode>
+          );
+          `,
+          "App.tsx":
+            `import * as React from 'react';
```

Comment:
If I'm understanding your whitespace issue correctly, could you use `prettier.format` on the code example so it's formatted in the stackblitz?

Reply:
I tried like:
   
   const formatCode = (code) => {
  return prettier.format(code, {
    parser: "typescript",
    plugins: [parserTypeScript],
  });
};

......
   
     "App.tsx": formatCode(
            `import * as React from 'react';
` +
              ComponentExample.sourceCode +
              `
          
          export default ${exampleName};
          `
          ),